msteams-platform/samples/virtual-assistant.md

```diff
@@ -0,0 +1,560 @@
+
+For more information refer [this](https://msteams-captain.visualstudio.com/xGrowth%20App%20Templates/_wiki/wikis/xGrowth.wiki/88/Virtual-Assistant-for-MS-Teams?anchor=rich-cards) section from this documentation.
+
+### Handle activities from group chat or channel scope
+
+Book-a-room bot is designed for private chats (personal/1:1 scope) only. Since we have customized Virtual Assistant to support group chat and channel scopes, the Virtual Assistant might be invoked from these scopes and thus, Book-a-room bot might get activities for the same. Hence Book-a-room bot is customized to handle those activities. The check has been put in `OnMessageActivityAsync` methods of Book-a-room bot's activity handler.
+
+```csharp
+    protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+    {
+        // Check if activities are from groupchat/ teams scope. This might happen when the bot is consumed by Virtual Assistant.
+        if (turnContext.Activity.Conversation.IsGroup == true)
+        {
+            await ShowNotSupportedInGroupChatCardAsync(turnContext).ConfigureAwait(false);
+        }
+        else
+        {
+            ...
+        }
+    }
+```
+
+You can also leverage existing skills from [Bot Framework Solutions repository](https://github.com/microsoft/botframework-solutions/tree/master/skills/csharp) or create a new skill altogether from scratch. Tutorials for the later can be found [here](https://microsoft.github.io/botframework-solutions/overview/skills/). Please refer to [documentation](https://docs.microsoft.com/azure/bot-service/skills-conceptual?view=azure-bot-service-4.0)   for Virtual Assistant and skills architecture.
+
+## Sample code to get started
+
+- [Updated visual studio template](https://github.com/nebhagat/msteams-virtual-assistant-dotnet)
+- [Book-a-room bot skill code](https://github.com/OfficeDev/microsoft-teams-apps-bookaroom/tree/nebhagat/microsoft-teams-apps-bookaroom-skill)
+
+## Virtual Assistant known limitations
+
+- **EndOfConversation**. A skill should send an `endOfConversation` activity when it finishes a conversation. basis this activity, a Virtual Assistant ends context with that particular skill and gets back into Virtual Assistant's (root) context. For Book-a-room bot, there is no clear state where conversation can be ended. Hence we have not sent `endOfConversation` from Book-a-room bot and when user wants to go back to root context they can simply do that by `start over` command.
+- **Card refresh**. Card refreshes is not yet supported Virtual Assistant.
+- **Messaging extensions**.:
+  - Currently, a Virtual Assistant can support a maximum of ten commands for messaging extensions.
+  - Configuration of message extensions is not scoped to individual commands but for the entire extension itself. This limits configuration for each individual skill through Virtual Assistant.
```

Comment:
"messaging extensions", not "message extensions"

Reply:
corrected.